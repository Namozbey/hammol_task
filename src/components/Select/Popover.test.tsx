import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { Popover } from ".";

const options = [
  { label: "All", action: () => console.log("all") },
  { label: "Docs", action: () => console.log("docs") },
];

describe("Testing Popover", () => {
  it("Snapshots", () => {
    const { asFragment } = render(
      <Popover options={options}>
        <button>click</button>
      </Popover>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Fire change event", () => {
    let isClicked = false;
    const action = () => {
      isClicked = true;
    };

    const newOptions = [
      { label: "All", action: () => console.log("all") },
      { label: "Docs", action },
    ];

    render(
      <Popover options={newOptions}>
        <button>expand</button>
      </Popover>
    );
    const element = screen.getByText("expand");
    fireEvent.click(element);
    fireEvent.click(screen.getByText("Docs"));

    expect(isClicked).toBe(true);
  });
});
