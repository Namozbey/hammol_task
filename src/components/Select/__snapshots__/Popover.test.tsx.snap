// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Popover Snapshots 1`] = `
<DocumentFragment>
  <div
    class="sc-gKXOVf idTazL"
  >
    <button>
      click
    </button>
    <div
      class="sc-iBkjds bEBgQR"
    >
      <label
        class="sc-eCYdqJ egtqok"
        font-size="16px"
      >
        <p>
          All
        </p>
      </label>
      <label
        class="sc-eCYdqJ egtqok"
        font-size="16px"
      >
        <p>
          Docs
        </p>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
